File: 0001-two-sum/0001-two-sum.c
#include <stdlib.h>

int* twoSum(int* nums, int numSize, int target, int* returnSize) {
    *returnSize = 2;  // The size of the result array is 2.
    int* result = (int*)malloc(2 * sizeof(int));  // Allocate memory for the result array.

    for (int i = 0; i < numSize; i++) {
        for (int j = i + 1; j < numSize; j++) {  // j should start from i+1 to avoid adding the same element.
            if (nums[i] + nums[j] == target) {
                result[0] = i;
                result[1] = j;
                return result;
            }
        }
    }

    // If no solution is found (this assumes there is always a solution based on the problem statement).
    *returnSize = 0;
    return NULL;
}

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int p[] = new int[2];
     for(int i=0;i<nums.length;i++)
     {
        int left = i+1;
        int right = nums.length-1;
        while(left<=right)
        {
            if((nums[i]+nums[left])==target)
            {
                p[0] = i;
                p[1]=left;
                return p;
            }
           else if((nums[i]+nums[right])==target)
            {
                p[0] = i;
                p[1]=right;
                return p;
            }
            else if(left!=right && (nums[left]+nums[right])==target)
            {
                p[0] = left;
                p[1]=right;
                return p;
            }
            left++;
            right--;
        }
     }   
     return p;
    }
}